public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -297,6 +321,25 @@ export function createDashboardSceneFromDashboardModel(oldModel: DashboardModel)
   return dashboardScene;
 }
 
+export function createVariableForSnapshots(variable: TypedVariableModel): SceneVariable {
+  const snapshotVariable = new SnapshotVariable({
```

Comment:
@axelavargas think you missed adding this file :)

Reply:
oh yes, before I was implemeting this on the Scenes repository. I moved the implementation here :)

Reply:
@axelavargas - I've just pushed import fix, otherwise this was not working :)

Anyways - seems like most of the variables are working now in snapshots, but not the ad hoc filters, the read only variable is empty even tho I have snapshoted a dashboard with filters applied.

Reply:
Thanks @dprokop 😄 , yes, I am working on adding support for ad-hoc filters, intervals and textBox variables :)

Reply:
@dprokop I have made a refactor to support `ad-hoc, intervals, groupBy, and textBox` variables. Displaying the variables works as expected, however, I see an issue with the URL and ad-hoc variables, the URL does not contain the value of filters 🤔.

![image](https://github.com/grafana/grafana/assets/239999/e5c61832-d6d9-410c-9548-643da8083f71)

I am unsure if this is related to my changes, will investigate. But for now, can you check the approach of this PR 😄.

Reply:
@axelavargas - i think it's because filters have a custom sync handler: https://github.com/grafana/scenes/blob/main/packages/scenes/src/variables/adhoc/AdHocFiltersVariable.tsx#L116

Reply:
@dprokop I managed to fix the issue with ad-hoc filters. I took the approach of using the `AdHocFiltersVariable` from scenes and adding the `read-only` prop to make it work.